### PR TITLE
Fix out of bound bug in _log function

### PR DESCRIPTION
**Pull request by szledan**
Signed-off-by: Szilard Ledan <szledan@gmail.com>

### CODE REVIEW

**Comment by kkristof** [comment]
LGTM

File: src/utils/gepard-defs.cpp [hunk 1]
```diff
@@ -33,9 +33,10 @@ namespace gepard {
 
 void _log(unsigned int level, const std::string& msg) {
   const int max_level = LOG_LEVEL;
-  if (level <= max_level) {
+  ASSERT(level);
+  if (level && level <= max_level) {
```
Comment (elecro) [inline src/utils/gepard-defs.cpp]: I think a better way would be to simply have the level argument to range from [0;2].

Comment (kkristof) [inline src/utils/gepard-defs.cpp]: If we would like finer grain logging (eg. `LOG5` for trapezoid builder), which is supported by the `LOGN` macro, then we didn't want to use ranges.

Comment (kkristof) [inline src/utils/gepard-defs.cpp]: I see, the color array is fixed.

**Comment by szledan** [comment]
Hi @elecro , I've updated the patch.

**Comment by dbatyai** [comment]
LGTM

**Pull request merged**